blocks/newsletter-modal/newsletter-modal.js

```diff
@@ -0,0 +1,101 @@
+  if (children.length < 4) return;
+
+  block.innerHTML = '';
+  const { createTag } = await import(`${getLibs()}/utils/utils.js`);
+
+  const picture = children[0].querySelector('picture');
+  const text = children[1].querySelector('div > p:nth-child(1)').textContent;
+  const disclaimer = children[1].querySelector('div > p:nth-child(2)').textContent;
+
+  const emailText = children[2].querySelector('div p:nth-child(1)').textContent;
+  const emailPlaceholder = children[2].querySelector('div p:nth-child(2)').textContent;
+  const cta = children[2].querySelector('div p:nth-child(3)').textContent;
+
+  const successMsg = children[3].querySelector(':scope > div:nth-child(1) ').textContent;
+  const errorMsg = children[3].querySelector(':scope > div:nth-child(2) ').textContent;
+
+  const bannerContainer = createTag('div', { class: 'newsletter-modal-banner-container' }, picture);
+  const content = createTag('div', { class: 'newsletter-modal-content' });
+
+  const textEl = createTag('p', { class: 'newsletter-modal-text' }, text);
+  const form = createTag('form', { class: 'newsletter-modal-form' });
+
+  const emailTextEl = createTag('span', { class: 'newsletter-modal-email-text', id: 'newsletter_email' }, emailText);
+  const emailEl = createTag('input', { type: 'email', class: 'newsletter-modal-email', required: 'required' });
+  const emailLabelEl = createTag('label', { class: 'newsletter-modal-email-label', for: 'newsletter_email' }, emailTextEl);
+  emailLabelEl.append(emailEl);
+  emailEl.placeholder = emailPlaceholder;
+
+  const ctaEl = createTag('button', { type: 'submit', class: 'newsletter-modal-cta' }, cta);
+  const disclaimerEl = createTag('p', { class: 'newsletter-modal-disclaimer' }, disclaimer);
+
+  ctaEl.addEventListener('click', (e) => {
+    e.preventDefault();
+    const email = emailEl.value;
+
+    if (email && emailEl.checkValidity()) {
+      const headers = new Headers();
+      headers.append('Content-Type', 'application/json');
+
+      const body = {
+        sname: 'adbeblog',
+        email,
+        consent_notice: '<div class="disclaimer detail-spectrum-m" style="letter-spacing: 0px; padding-top: 15px;">The Adobe family of companies may keep me informed with personalized emails from the Adobe Blog team. See our <a href="https://www.adobe.com/privacy/policy.html" target="_blank">Privacy Policy</a> for more details or to opt-out at any time.</div>',
+        current_url: window.location.href,
+      };
+
+      const requestOptions = {
+        method: 'POST',
+        headers,
+        body: JSON.stringify(body),
+      };
+
+      fetch('https://www.adobe.com/api2/subscribe_v1', requestOptions)
+        .then(() => {
+          displayConfirmation(content, successMsg, createTag);
+        })
+        .catch(() => {
+          displayConfirmation(content, errorMsg, createTag);
+        });
+    } else {
+      emailEl.classList.add('error');
+      emailEl.reportValidity();
+    }
+  });
```

Comment:
I think the listener's function should be pulled out. It adds a lot of bulk to `init`